src/cpp/process_image.cpp

```diff
@@ -3,58 +3,91 @@
 #include <filesystem>
 #include <string>
 
+#include "filesystem.hpp"
 #include "process_image.hpp"
 
+// This is to make sure that the opencv functions are not
+// called from android but work in ios
+#ifdef __APPLE__
+#include "filter_image.hpp"
+#else
+#define filterImagePre ERROR_NO_ANDROID
+#endif
+
```

Comment:
Making sure filter_image.hpp is only included if openCV is defined

Reply:
The logic makes sense but how do you know OpenCV is defined if __APPLE__ is defined?

Reply:
I know openCV works on iOS, but you are right it would’ve been better to use #ifdef cv instead